aiogram/api/client/session/aiohttp.py

```diff
@@ -1,29 +1,72 @@
 from __future__ import annotations
 
-from typing import AsyncGenerator, Callable, Optional, TypeVar, cast
+from typing import AsyncGenerator, Callable, Optional, TypeVar, Tuple, Dict, Any, Union, cast
 
-from aiohttp import ClientSession, ClientTimeout, FormData
+from aiohttp import ClientSession, ClientTimeout, FormData, BasicAuth, TCPConnector
 
 from aiogram.api.methods import Request, TelegramMethod
 
 from .base import PRODUCTION, BaseSession, TelegramAPIServer
 
 T = TypeVar("T")
+_ProxyType = Union[str, Tuple[str, BasicAuth]]
 
 
-class AiohttpSession(BaseSession):
+class AiohttpSession(BaseSession[_ProxyType]):
     def __init__(
         self,
         api: TelegramAPIServer = PRODUCTION,
         json_loads: Optional[Callable] = None,
         json_dumps: Optional[Callable] = None,
+        proxy: Optional[_ProxyType] = None,
     ):
-        super(AiohttpSession, self).__init__(api=api, json_loads=json_loads, json_dumps=json_dumps)
+        super(AiohttpSession, self).__init__(
+            api=api,
+            json_loads=json_loads,
+            json_dumps=json_dumps,
+            proxy=proxy
+        )
         self._session: Optional[ClientSession] = None
+        self.cfg.connector_type = TCPConnector
+        self.cfg.connector_init = cast(Dict[str, Any], {})
+
+        if self.proxy:
+            try:
+                from aiohttp_socks import ProxyConnector
+                from aiohttp_socks.utils import parse_proxy_url
+            except ImportError as exc:
+                raise UserWarning(
+                    "In order to use aiohttp client for proxy requests, install "
+                    "https://pypi.org/project/aiohttp-socks/"
+                ) from exc
+
+            if isinstance(self.proxy, str):
+                proxy_url, proxy_auth = self.proxy, None
+            else:
+                proxy_url, proxy_auth = self.proxy
+
+            self.cfg.connector_type = ProxyConnector
+
+            proxy_type, host, port, username, password = parse_proxy_url(proxy_url)
+            if proxy_auth:
+                if not username:
+                    username = proxy_auth.login
+                if not password:
+                    password = proxy_auth.password
+
+            self.cfg.connector_init.update(
+                dict(
+                    proxy_type=proxy_type, host=host, port=port,
+                    username=username, password=password,
+                    rdns=True,
```

Comment:
What is rdns? Should we give the ability to change this param by user?

Reply:
`rdns` is `use no resolver` boolean. Current aiogram-dev-2.x implements it this way.

Reply:
Thank you for your answer!